database/sinkdb/state.go

```diff
@@ -90,6 +90,13 @@ func (s *state) RestoreSnapshot(data []byte, index uint64) error {
 	s.peers = snapshot.Peers
 	s.state = snapshot.State
 	s.version = snapshot.Version
+
+	if s.version == nil {
+		s.version = make(map[string]uint64)
+	}
+	if s.peers == nil {
+		s.peers = make(map[uint64]string)
+	}
```

Comment:
I don't think we need them because
1. no snapshot should ever have 0 peers. The manually triggered snapshots could've caused that to happen if the snapshot was taken before the initial conf change entries were applied.
2. a snapshot should always at least have a version for the nextNodeID.

We might be violating that second invariant if you create a brand new cluster and don't use it at all. Maybe instead of doing the version nil check, we should initialize the version map with a version for the nextNodeID in [newState](https://github.com/chain/chain/blob/main/database/sinkdb/state.go#L35).

Something like
```
		version: map[string]uint64{nextNodeID: 0},
```